lib/projects/addCredits.ts

```diff
@@ -0,0 +1,76 @@
+import { NetworkType, getNetworkConfig } from './utils/getNetworkConfig'
+import CreditManagerAbi from '../abis/CreditManager.json'
+import arg from 'arg';
+import { PROJECT_INFO } from './Projects';
+
+if (!process.env.PRIVATE_KEY) {
+  throw new Error('PRIVATE_KEY environment variable is not set');
+}
+
+
+const getCreditInfo = (creditId: string) => {
+  const info = PROJECT_INFO.find(projectInfo => `${projectInfo.projectId}-${projectInfo.vintage}` === creditId)
+  
+  if (!info) {
+    throw(Error(`Credit id ${creditId} not found`))
+  }
+
+  return [
+    info.address,
+    info.tokenId,
+    info.isExAnte,
+    info.projectId,
+    info.vintage,
+    info.name,
+    [info.methodology],
+    info.category,
+    info.country,
+    "", // region
+  ]
+}
+
+const addCredits = async (network: string, creditIds: string[]) => {
+  const { getContract, addresses } = getNetworkConfig(network as NetworkType)
+  
+  const credits = creditIds.map(creditId => getCreditInfo(creditId))
+
+  const creditManagerContractAddress = addresses.creditManager.address
+  
+  const creditManagerContract = getContract(creditManagerContractAddress, CreditManagerAbi)
+
+  const batchSize = 10
+  const numberOfBatches = Math.ceil(creditIds.length / batchSize)
+  
+  for (let i = 0; i < numberOfBatches; i++) {
+    const start = i * batchSize
+    const end = start + batchSize
+    const currentBatch = credits.slice(start, end)
+
+    try {
+      const tx = await creditManagerContract.addCreditBatch(currentBatch)
```

Comment:
Can you remind me how we plan to manage this contract / address / PK?
It would be nice if we upgraded it to have AccessControl so we can grant and revoke write powers to multiple systems, like retool. For now the plan is just to have someone hold the EOA pk and run this script whenever we need to push project data?